enhancements/declarative-index-config.md

```diff
@@ -725,45 +983,59 @@ $ cat community-operators/etcd.json
         "base64data":"iVBORw0KGgoAAAANSUhEUgAAA.....",
         "mediatype":"image/png"
     },
-    "channels": ["alpha"],
     "description": "A message about etcd operator, a description of channels"
-},
+}
 {
-    "schema": "olm.bundle", 
+    "schema": "olm.bundle",
     "name": "etcdoperator-community.v0.6.1",
     "package": "etcd",
     "image": "quay.io/operatorhubio/etcd:v0.6.1",
-    "version": "0.6.1",
     "properties":[
         {
-            "name": "olm.package",
-            "values": {
+            "type": "olm.package",
+            "value": {
                 "packageName": "etcd",
                 "version": "0.6.1"
             }
         },
         {
-            "name":"olm.gvk",
-            "type": "provided",
-            "values":{
-                "group": "etcd.database.coreos.com", 
-                "kind": "EtcdCluster", 
+            "type": "olm.package.provided",
+            "value": {
+                "packageName": "etcd",
+                "version": "0.6.1"
+            }
+        },
+        {
+            "type": "olm.gvk",
+            "value": {
+                "group": "etcd.database.coreos.com",
+                "kind": "EtcdCluster",
+                "version": "v1beta2"
+            }
+        },
+        {
+            "type": "olm.gvk.provided",
+            "value": {
+                "group": "etcd.database.coreos.com",
+                "kind": "EtcdCluster",
                 "version": "v1beta2"
             }
         },
         {
-            "name":"olm.channel",
-            "value": "alpha"
+            "type": "olm.channel",
+            "value": {
+                "name": "alpha"
+            }
         }
     ],
     "relatedImages": [
         {
-        "name": "etcdv0.6.1",
-        "image": "quay.io/coreos/etcd-operator@sha256:bd944a211eaf8f31da5e6d69e8541e7cada8f16a9f7a5a570b22478997819943"
+            "name": "etcdv0.6.1",
+            "image": "quay.io/coreos/etcd-operator@sha256:bd944a211eaf8f31da5e6d69e8541e7cada8f16a9f7a5a570b22478997819943"
         }
     ]
 }
-$ opm index add --bundles quay.io/operatorhubio/etcd:v0.9.0 --mode replaces --config community-operators/etcd
+$ opm index add --bundles quay.io/operatorhubio/etcd:v0.9.0 --mode replaces
```

Comment:
Seems like this command is incomplete?

Reply:
Also it should probably be `opm registry add --bundle quay.io/operatorhubio/etcd:v0.9.0 --mode replaces comunnity-operators/etcd` 
(to continue @njhale's suggestion about required inputs being unnamed arguments)
Totally cool if we want to tackle this in a separate update PR though

Reply:
The underlying db representation is opaque to the index commands, right?

The `opm index add` command will build a declarative config `olm.bundle` blob for each bundle passed to `--bundles`. One of the fields in `olm.bundle` is `package`, which we'll set. Since we know the package name, I'm not sure we need to be told about it on the command line.

It should be possible to add multiple bundles from different packages as well.

Reply:
And just to clear up one bit of confusion that I missed in your comment the first time, this is `opm index add`, not `opm registry add`.

For `opm registry add`, I would expect `opm registry add --bundles quay.io/operatorhubio/etcd:v0.9.0,quay.io/operatorhubio/prometheus:v1.2.3 --mode replaces --database community-operators`

And `--database` also accepts an sqlite database or a tarball, and we'll figure out which it is and do the right thing.

Reply:
So the section starts off with: 
```
The opm index add command will also be enhanced to create olm.bundle json blobs for bundles passed onto the command along with a package config file, and append them to the config file.
```

and then goes on to describe how `opm index add` will take a directory of configs as input, and output a directory of configs with the bundles in `--bundles` added in. 

This update looks like 
a) Is removing the operand it's supposed to operate on 
b) highlights an issue previously brought up by @njhale: what is the expected input operand of `opm index` commands? Is it a container image? Or is it a folder? Should it be both? Why not let `opm regsitry add` operate on the directory of configs instead?

Reply:
My impression is that `index` subcommands exclusively work against an index image, and `registry` subcommands exclusively work against the database. So since declarative config is effectively a database format, only `registry` subcommands should take configs as input.

`opm index add` is effectively:

Reply:
My impression is that `index` subcommands exclusively work against an index image, and `registry` subcommands exclusively work against the database. So since declarative config is effectively a database format, only `registry` subcommands should take configs as input.

`opm index add $@` is effectively:
1. $container_tool pull $index_image
2. $container_tool cp /path/to/index.db .
3. opm registry add --database index.db $@
4. (rebuild image with updated index.db)

Same for each of the other subcommands.

What this means is that the `index` subcommands specifically do NOT take the database as input because it auto-detects its location in the index image via a label, extracts it, modifies it, then rebuilds.

The `opm registry *` commands will take the configs as input.

Reply:
I updated the section text from `opm index add` to `opm registry add`, and changed the command line to use `--database` instead of `--configs` since we aren't (yet at least) introducing new config-specific flags on existing commands.